logging/lib/src/extension.dart

```diff
@@ -0,0 +1,54 @@
+// ignore_for_file: public_member_api_docs
+
+import 'package:logging/logging.dart';
+import 'package:sentry/sentry.dart';
+import 'package:sentry/sentry_io.dart';
+
+extension LogRecordX on LogRecord {
+  Breadcrumb toBreadcrumb() {
+    return Breadcrumb(
+      category: 'log',
+      type: 'debug',
+      timestamp: time,
+      level: level.toSentryLevel(),
+      message: message,
+      data: <String, Object>{
+        if (object != null) 'LogRecord.object': object!,
+        if (error != null) 'LogRecord.error': error!,
+        if (stackTrace != null) 'LogRecord.stackTrace': stackTrace!,
+        'LogRecord.loggerName': loggerName,
+        'LogRecord.sequenceNumber': sequenceNumber,
+      },
+    );
+  }
+
+  SentryEvent toEvent() {
+    return SentryEvent(
+      logger: loggerName,
+      level: level.toSentryLevel(),
+      message: SentryMessage(message),
```

Comment:
@ueman before merging this, a question:
is it possible that people call `message: 'a simple message number $number'`, right?

the problem with this is, Sentry groups messages (when no stack trace is given), by the message String.

So a message `a simple message number 1` and `a simple message number 2` won't be grouped together.

is there a way to detect that and replace by the message `template` and `params`?

eg:
`template: 'a simple message number %d'`,
`params: ['1']`

and 

`template: 'a simple message number %d'`,
`params: ['2']`

both messages would be grouped together on Sentry, is that feasible?

Reply:
> is that feasible?

No, unfortunately that's not possible.

Reply:
yeah I suspected so, it'd be fine if the logging library actually have such template and params

Reply:
The logging library doesn't have that either :(

Reply:
We can work with what https://pub.dev/documentation/logging/latest/logging/LogRecord-class.html gives us.